sparkle/src/components/Checkbox.tsx

```diff
@@ -1,122 +1,98 @@
+import * as CheckboxPrimitive from "@radix-ui/react-checkbox";
+import { cva, VariantProps } from "class-variance-authority";
 import React from "react";
 
 import { CheckIcon, DashIcon } from "@sparkle/icons/solid";
-import { classNames } from "@sparkle/lib/utils";
+import { cn } from "@sparkle/lib/utils";
 
 import { Icon } from "./Icon";
+import { Label } from "./Label";
 
-export type CheckBoxCheckedStatus = "checked" | "unchecked" | "partial";
-
-export interface CheckboxProps {
-  variant?: "selectable" | "checkable";
-  size?: "xs" | "sm";
-  checked?: CheckBoxCheckedStatus;
-  disabled?: boolean;
-  onChange: (checked: boolean) => void;
-  className?: string;
-}
-
-export function Checkbox({
-  variant = "selectable",
-  checked = "unchecked",
-  size = "sm",
-  onChange,
-  className = "",
-  disabled,
-}: CheckboxProps) {
-  let isChecked: boolean;
-  let isPartialChecked: boolean;
-  switch (checked) {
-    case "checked":
-      isChecked = true;
-      isPartialChecked = false;
-      break;
-    case "unchecked":
-      isChecked = false;
-      isPartialChecked = false;
-      break;
-    case "partial":
-      isChecked = true;
-      isPartialChecked = true;
-      break;
-    default:
-      throw new Error("Invalid checked prop");
+const checkboxStyles = cva(
+  cn(
+    "s-shrink-0 s-peer s-border s-border-primary-500 s-text-foreground",
+    "data-[state=checked]:s-text-white data-[state=checked]:s-text-foreground",
+    "focus-visible:s-ring-ring s-ring-offset-background focus-visible:s-outline-none focus-visible:s-ring-2 focus-visible:s-ring-offset-2",
+    "disabled:s-cursor-not-allowed disabled:s-opacity-50"
+  ),
+  {
+    variants: {
+      isPartial: {
+        false: "data-[state=checked]:s-bg-action-500",
+        true: "data-[state=checked]:s-bg-element-700",
+      },
+      size: {
+        xs: "s-h-4 s-w-4 s-rounded",
+        sm: "s-h-5 s-w-5 s-rounded-md",
+      },
+    },
+    defaultVariants: {
+      size: "sm",
+      isPartial: false,
+    },
```

Comment:
Not sure to be a big fan of `cva` and `VariantProps` I find the props definition less explicit.. What do you think @flvndvd @tdraier ?

Reply:
It's actually a pretty good pattern. For design systems where the main person that will update the style is a designer, it's way simpler than trying to implement conditional code down the line.